src/_imaging.c

```diff
@@ -968,95 +976,120 @@ _convert_transparent(ImagingObject *self, PyObject *args) {
+HPyDef_METH(Imaging_copy, "copy", Imaging_copy_impl, HPyFunc_NOARGS)
+static HPy Imaging_copy_impl(HPyContext *ctx, HPy self) {
+    PyObject *py_self = HPy_AsPyObject(ctx, self);
+    Imaging im = PyImaging_AsImaging(py_self);
+    return HPy_FromPyObject(ctx, PyImagingNew(ImagingCopy(im)));
 }
 
-static PyObject *
-_crop(ImagingObject *self, PyObject *args) {
+
+HPyDef_METH(Imaging_crop, "crop", Imaging_crop_impl, HPyFunc_VARARGS)
+static HPy Imaging_crop_impl(HPyContext *ctx, HPy self, HPy *args, HPy_ssize_t nargs) {
     int x0, y0, x1, y1;
-    if (!PyArg_ParseTuple(args, "(iiii)", &x0, &y0, &x1, &y1)) {
-        return NULL;
+    HPy h_tuple;
+    PyObject *tuple;
+
+    if (!HPyArg_Parse(ctx, NULL, args, nargs, "O", &h_tuple)) {
+        return HPy_NULL;
     }
 
-    return PyImagingNew(ImagingCrop(self->image, x0, y0, x1, y1));
+    tuple = HPy_AsPyObject(ctx, h_tuple);
+    
+    x0 = PyLong_AsLong(PyTuple_GetItem(tuple, 0));
+    y0 = PyLong_AsLong(PyTuple_GetItem(tuple, 1));
+    x1 = PyLong_AsLong(PyTuple_GetItem(tuple, 2));
+    y1 = PyLong_AsLong(PyTuple_GetItem(tuple, 3));
+
+    PyObject *py_self = HPy_AsPyObject(ctx, self);
+    Imaging im = PyImaging_AsImaging(py_self);
+
+    return HPy_FromPyObject(ctx, PyImagingNew(ImagingCrop(im, x0, y0, x1, y1)));
 }
 
-static PyObject *
-_expand_image(ImagingObject *self, PyObject *args) {
+HPyDef_METH(Imaging_expand_image, "expand_image", Imaging_expand_image_impl, HPyFunc_VARARGS)
+static HPy Imaging_expand_image_impl(HPyContext *ctx, HPy self, HPy *args, HPy_ssize_t nargs) {
+    PyObject *py_self = HPy_AsPyObject(ctx, self);
+    Imaging im = PyImaging_AsImaging(py_self);
     int x, y;
     int mode = 0;
-    if (!PyArg_ParseTuple(args, "ii|i", &x, &y, &mode)) {
-        return NULL;
+    if (!HPyArg_Parse(ctx, NULL, args, nargs, "ii|i", &x, &y, &mode)) {
+        return HPy_NULL;
     }
-
-    return PyImagingNew(ImagingExpand(self->image, x, y, mode));
+    return HPy_FromPyObject(ctx, PyImagingNew(ImagingExpand(im, x, y, mode)));
 }
 
-static PyObject *
-_filter(ImagingObject *self, PyObject *args) {
+HPyDef_METH(Imaging_filter, "filter", Imaging_filter_impl, HPyFunc_VARARGS)
+static HPy Imaging_filter_impl(HPyContext *ctx, HPy self, HPy *args, HPy_ssize_t nargs) {
     PyObject *imOut;
     Py_ssize_t kernelsize;
     FLOAT32 *kerneldata;
 
+    PyObject *py_self = HPy_AsPyObject(ctx, self);
+    Imaging im = PyImaging_AsImaging(py_self);
+
     int xsize, ysize, i;
     float divisor, offset;
-    PyObject *kernel = NULL;
-    if (!PyArg_ParseTuple(
-            args, "(ii)ffO", &xsize, &ysize, &divisor, &offset, &kernel)) {
-        return NULL;
+    HPy kernel = HPy_NULL, h_size;
+
+    if (!HPyArg_Parse(
+            ctx, NULL, args, nargs, "OffO", &h_size, &divisor, &offset, &kernel)) {
+        return HPy_NULL;
     }
 
+    PyObject *py_size = HPy_AsPyObject(ctx, h_size);
+    PyObject *py_kernel = HPy_AsPyObject(ctx, kernel);
```

Comment:
Note that `HPy_AsPyObject` increments ref count before returning the underlying `PyObject*` to you, so you should `Py_DECREF(py_kernel)` when you're done with it (or leave it be and fix it by migrating it all to HPy).